spec/services/variants_stock_levels_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+describe VariantsStockLevels do
+  let(:order) { create(:order) }
+  let(:product) { create(:simple_product) }
+
+  let!(:line_item) do
+    create(:line_item, order: order, variant: variant_in_the_order, quantity: 2, max_quantity: 3)
+  end
+  let!(:variant_in_the_order) { create(:variant, count_on_hand: 4) }
+  let!(:variant_not_in_the_order) { create(:variant, count_on_hand: 2) }
+
+  let(:variant_stock_levels) { VariantsStockLevels.new }
+
+  before do
+    order.reload
+  end
```

Comment:
Have you tried removing it now that tests are finished? We modify an order object from subsequent lets in several tests and they don't reload it. Why this case would be different?

Reply:
I copied this from the original test and now I tried removing it but it is needed @sauloperez 
I am not sure why but it must be because of some relation that is not set explicit in the factory and is created only on reload.
But I found an unnecessary line! let(:product) :-D I removed that line and rebased.

Reply:
great! at least we removed code. It must definitely be one of those callbacks...